src/Sylius/Bundle/CoreBundle/Checker/CatalogPromotionEligibilityChecker.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\Checker;
+
+use Sylius\Calendar\Provider\DateTimeProviderInterface;
+use Sylius\Component\Core\Model\CatalogPromotionInterface;
+use Sylius\Component\Promotion\Model\CatalogPromotionStates;
+
+final class CatalogPromotionEligibilityChecker implements CatalogPromotionEligibilityCheckerInterface
+{
+    public function __construct(private DateTimeProviderInterface $dateTimeProvider)
+    {
+    }
+
+    public function isCatalogPromotionEligible(CatalogPromotionInterface $promotion): bool
+    {
+        return $promotion->isEnabled() && $promotion->getState() !== CatalogPromotionStates::STATE_FAILED;
```

Comment:
What should happen to the promotion that has been failed? What does it even mean? How can it occur? Can we anyhow reapply it?

Reply:
Also, perhaps it would be better to check if we can make a transition to a given state (like active) instead of hardcoding state different then failed

Reply:
Ok, I've changed the way I checking it.